examples/3.routes/8.tolls-and-ferries/interface.js

```diff
@@ -0,0 +1,260 @@
+import { getDurationString } from '../../../utils';
+
+// Array used for journey overview.
+let journeyLegs = [];
+let wholeRouteContainsTolls = false;
+let wholeRouteContainsFerries = false;
+
+/**
+ * Render journey overview.
+ *
+ * @param data {object} route specification
+ */
+export const renderRouteData = data => {
+  // the total distance of the route, in meters converted to km
+  const routeDistance = data.distance ? `${(data.distance / 1000).toFixed(0)} km` : 'Unknown';
+
+  // the total energy used of the route, in kWh
+  const routeEnergy = data.consumption ? `${data.consumption.toFixed(2)} kWh` : 'Unknown';
+
+  // the total duration of the journey (including charge time), in seconds
+  document.getElementById('duration').innerHTML = `${getDurationString(data.duration ?? 0)}`;
+
+  let extraFerryStops = 0;
+  let origin = '';
+  let destination = '';
+  // Convert the legs we receive from the API to journey overview legs we can display.
+  // First check if the route has loaded yet
+  if (data) {
+    // Determine origin and destination
+    origin = data.legs[0].origin.properties.name;
+    destination = data.legs[data.legs.length - 1].destination.properties.name;
+    // Go through the legs we receive from the backend - stations and final.
+    for (let i = 0; i < data.legs.length; i++) {
+      let ferrySteps = getNumberOfFerryStepsInLeg(data.legs[i]);
+      // Used to determine number of stops, at top of journey overview details
+      extraFerryStops += ferrySteps;
+      // 'Regular' route, no ferry. Add route to station as a leg
+      if (ferrySteps == 0) {
```

Comment:
to see if a leg contains a toll or a ferry please use the tag property on the leg level. Then you can skip counting if the tag is not there.

Reply:
this is also used at the top of the journey overview to calculate the number of stops:
```
[67] const routeStops = `${data.charges + extraFerryStops ?? 0} stops`;
```
I would argue here that a stop should consist of either a charging station (no calculation necessary as this is in route.charges), as well as the stop(s) to 'get on' to a ferry

Reply:
Yes that is true. But if the whole leg does not contain a ferry the count is not necessary. However when it does then you would need to do so.

Reply:
good point! will modify so it only counts when it's present